### ISSUE TITLE
The future of the Metadatabus

### ISSUE THREAD

**Issue by RussellMcOrmond** [crkn-rcdr/Access-Platform#92]
See: [Canadiana Platform II (current) and III (in development): metadata bus](https://crkn.sharepoint.com/:w:/s/platform/EeBxfNgPK8BOjC87E06vdlwBcnFYPU4kWLnQ1hfSte6tLw?e=vAuzHX)   , which also links to [a document describing the older micro services](https://crkn.sharepoint.com/:w:/s/platform/EWMRykJo66pEpHuf3Y7kh3EBfyc2lQ3vjFC4P2GVrMuwNQ?e=jMCl6u).

This issue will be used to discuss the old metadata bus, and allow us to think towards the future.

**Comment by RussellMcOrmond** [comment]
Some updates:

* `manifest` and `collection` databases have been merged, and is where most of the current work is happening.
* We are mid-way through some of the "stages" mentioned on page 4.
  * We have an `internalmeta2`, `copresentation2`, and `cosearch2` that mimic the schemas used by the old platform as an intermediary step.
   * `copresentation2` can be used by the current version of https://github.com/crkn-rcdr/cap .
   *  `cosearch` is streamed to Solr, using the same Solr schema as we have been using, and CAP accesses that Solr core.
   * Hammer2 exists, currently written in Perl in https://github.com/crkn-rcdr/CIHM-Meta
   * Demo portals are all live, ponting at the *2 databases on CouchDB and Solr:
      * http://www-demo.canadiana.ca/
      * http://heritage-demo.canadiana.ca/
      * http://gac-demo.canadiana.ca/
      * http://mcgillarchives-demo.canadiana.ca/
      * http://nrcan-demo.canadiana.ca/
      * http://numeris-demo.canadiana.ca/
      * http://parl-demo.canadiana.ca/
      * http://pub-demo.canadiana.ca/
      * http://sve-demo.canadiana.ca/

**Comment by RussellMcOrmond** [comment]
Suggestion for replacement of "SolrStream".

What we currently do is have Hammer(2) output documents to `cosearch` and `copresentation`.  CAP uses `copresentation` directly.  `cosearch` is replicated to all application servers, and then is streamed to each Solr instance we have (We have 4, two in each of Edmonton and Toronto).

This involves inefficient storage in order to have a simplistic SolrStream tool which simply copies from CouchDB to Solr https://github.com/crkn-rcdr/CIHM-Meta/blob/main/lib/CIHM/Meta/SolrStream.pm

If instead we created different views (or an index) that allowed us to ask "Which Solr instance has been updated for this record", we could avoid that duplication and complexity.

This relates to https://github.com/crkn-rcdr/Access-Platform/issues/91 , which is a reminder to normalise and validate our data earlier in our processing so we don't have database errors at late stages.
